
Пример.
public class ExceptionExample {
    public static void main(String[] args) {
        try {
            int result = 10 / 0;  // Деление на ноль вызывает исключение
        } catch (ArithmeticExcetion e) {  // Ловим исключение деления на ноль
            System.out.println("Ошибка: Деление на ноль!");
        } finally {
            System.out.println("Блок finally всегда выполнится.");
        }
    }
}

Наследуются от КЛАССА Throwable на 2 ветки ERROR и EXCEPTION

OBJECT -> THROWABLE -> ERROR/EXCEPTION

Бывают проверяемые и непроверяемые.

Проверяемые исключения
Проверяемые исключения — это исключения, которые Java обязана обработать или "передать" дальше.
Java требует, чтобы такие исключения были обработаны с помощью try-catch или указаны
в сигнатуре метода с использованием ключевого слова throws.

Непроверяемые исключения
Непроверяемые исключения — это исключения, которые не требуют обязательной обработки.
Они представляют собой ошибки программирования, такие как NullPointerException, ArrayIndexOutOfBoundsExceptio...
Возникают по ВИНЕ ПРОГРАММИСТА.

Отличие: Проверяемые исключения требют обязательной обработки или указания в методе с помощью throws.

кл. слова: try catch finally throw, throws

try — блок, в котором может произойти исключение.
catch — блок, который перехватывает и обрабатывает исключение, если оно произошло в блоке try.
finally — блок, который всгда выполнеятся после try-catch,
независимо от того, произошло исключение или нет.
throw — используется для явного выбрасывания исключения.
throws — используется в сигнатуре метода для указания, что этот метод может выбросить исключение.